includes/class-wp-job-manager-post-types.php

```diff
@@ -461,6 +462,35 @@ public function job_feed() {
 		remove_filter( 'posts_search', 'get_job_listings_keyword_search' );
 	}
 
+	/**
+	 * Modifies WordPress Query of public search.
+	 *
+	 * @param WP_Query $query Query being processed.
+	 */
+	public function public_search_handler( $query ) {
+		if ( ! $query->is_search() ) {
+			return;
+		}
+
+		// Remove filled positions, if necessary.
+		if ( 1 === absint( get_option( 'job_manager_hide_filled_positions' ) ) ) {
+			$meta_query = $query->get( 'meta_query' );
+			if ( ! is_array( $meta_query ) ) {
+				$meta_query = array();
+			}
+
+			$meta_query[] = [
```

Comment:
WPJM currently requires compatibility with PHP 5.2. Short-array syntax is currently not available.

Reply:
Oh, you're right, sorry @jom. I've made a commit with the necessary change.